### PR TITLE
feat: Add IDH Argo/Argo Events

### PR DESCRIPTION
**Pull request by tumido**
## This Pull Request implements

Migrate DH argo deployments from Ansible to Argo CD/kustomize using the same manifests already used by `aiops-prod-argo`. 

:warning: Requires manual cleanup in the namespaces first probably. :warning: 

## If migrating an Application to ArgoCD
- [x] I have completed the list of action items on [this page](https://aicoe.github.io/aicoe-cd/get_argocd_to_manage_your_app/)

### CODE REVIEW

**Comment by tumido** [comment]
cc @anishasthana

File: manifests/overlays/prod/applications/data_hub/dh-prod-argo.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+spec:
+  destination:
+    namespace: dh-prod-argo
+    server: https://datahub.psi.redhat.com:443
+  project: data-hub
+  source:
+    path: argo/overlays/dh-prod-argo
+    repoURL: https://github.com/AICoE/idh-manifests.git
+    targetRevision: production
+  syncPolicy:
```
Comment (anishasthana) [inline manifests/overlays/prod/applications/data_hub/dh-prod-argo.yaml:14]: Could we disabled auto-sync initially? Just until we have ArgoCD fully managing things.

File: manifests/overlays/prod/applications/data_hub/dh-prod-argo.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  destination:
+    namespace: dh-prod-argo
+    server: https://datahub.psi.redhat.com:443
+  project: data-hub
+  source:
+    path: argo-events/overlays/dh-prod-argo
+    repoURL: https://github.com/AICoE/idh-manifests.git
+    targetRevision: production
+  syncPolicy:
+    automated:
```
Comment (anishasthana) [inline manifests/overlays/prod/applications/data_hub/dh-prod-argo.yaml:35]: same as above

**Review by anishasthana — CHANGES_REQUESTED** [review]

File: manifests/overlays/prod/applications/data_hub/dh-prod-argo.yaml (hunk 1 above)
Comment (tumido) [reply manifests/overlays/prod/applications/data_hub/dh-prod-argo.yaml:14]: You mean, disable auto-sync in prod and have it enabled for stage only? That make sense. :+1: Will do. :slightly_smiling_face:

File: manifests/overlays/prod/applications/data_hub/kustomization.yaml [hunk 3]
```diff
@@ -17,3 +17,5 @@ resources:
 - dh-prod-customer-journeys.yaml
 - dh-stage-superset.yaml
 - dh-prod-superset.yaml
+- dh-stage-argo.yaml
+# - dh-prod-argo.yaml  # FIXME: Unlisted until dh-stage-argo verifies the migration success
```
Comment (tumido) [inline manifests/overlays/prod/applications/data_hub/kustomization.yaml:21]: @anishasthana @gmfrasca unlisted! :smirk:

**Review by anishasthana — APPROVED** [review]
:ship:

**Comment by tumido** [comment]
@anishasthana rebased

**Review by anishasthana — APPROVED** [review]

**Review by gmfrasca — APPROVED** [review]
:pizza:

**Comment by anishasthana** [comment]
@tumido can we unhold this PR?

**Pull request merged**